dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/scheduling/JobSchedulerController.java

```diff
@@ -0,0 +1,274 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.Value;
+
+import org.hisp.dhis.common.IdentifiableObject;
+import org.hisp.dhis.common.OpenApi;
+import org.hisp.dhis.feedback.ConflictException;
+import org.hisp.dhis.feedback.NotFoundException;
+import org.hisp.dhis.scheduling.JobConfiguration;
+import org.hisp.dhis.scheduling.JobConfigurationService;
+import org.hisp.dhis.scheduling.JobQueueService;
+import org.hisp.dhis.scheduling.JobStatus;
+import org.hisp.dhis.scheduling.JobType;
+import org.hisp.dhis.scheduling.SchedulingType;
+import org.hisp.dhis.webapi.openapi.SchemaGenerators.UID;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * API for scheduler list and named queues (sequences).
+ *
+ * @author Jan Bernitt
+ */
+@OpenApi.Tags( "system" )
+@RestController
+@RequestMapping( value = "/scheduler" )
+@AllArgsConstructor
+public class JobSchedulerController
+{
+    private final JobConfigurationService jobConfigurationService;
+
+    private final JobQueueService jobQueueService;
+
+    @Value
+    static class SchedulerEntry
+    {
+        @JsonProperty
+        String name;
+
+        @JsonProperty
+        String type;
+
+        @JsonProperty
+        String cronExpression;
+
+        @JsonProperty
+        Date nextExecutionTime;
+
+        @JsonProperty
+        JobStatus status;
+
+        @JsonProperty
+        boolean enabled;
+
+        @JsonProperty
+        boolean configurable;
+
+        @JsonProperty
+        List<SchedulerEntryJob> sequence;
+
+        static SchedulerEntry of( JobConfiguration config )
+        {
+            return new SchedulerEntry(
+                config.getName(),
+                config.getJobType().name(),
+                config.getCronExpression(),
+                config.getNextExecutionTime(),
+                config.getJobStatus(),
+                config.isEnabled(),
+                config.getJobType().isConfigurable(),
+                List.of( SchedulerEntryJob.of( config ) ) );
+        }
+
+        static SchedulerEntry of( List<JobConfiguration> queue )
+        {
+            JobConfiguration trigger = queue.get( 0 );
+            if ( trigger.getQueueName() == null )
```

Comment:
Perhaps add a method with Javadoc to `JobConfiguration` to encapsulate the null check and have the method explicitly indicate the meaning of this property being null?

Reply:
I added a `isUsedInQueue()`

Reply:
Looks great.